torch/utils/cpp_extension.py

```diff
@@ -142,15 +144,17 @@ def _find_rocm_home() -> Optional[str]:
     return rocm_home
 
 def _find_sycl_home() -> Optional[str]:
```

Comment:
@functools.lru_cache(maxsize=1)

Reply:
the _find_sycl_home, similar with _find_cuda_home is only called only one time and cached in SYCL_HOME.